### PR TITLE
Fix K generator in ECDSA

### PR DESCRIPTION
**Pull request by fanatid**
This bug gives different signatures for same transactions. This means that you got different txIds for same data.

/cc @killerstorm

### CODE REVIEW

**Comment by matiu** [comment]
Related + some Test cases: https://github.com/bitpay/bitcore/issues/1269

**Comment by braydonf** [comment]
I see, so the issue is that the sighash is reversed when it's signed but not when generating k, and leading to discrepancies between libraries in the situations that "little" endian is used.

Also verified against libsecp256k1 and with this PR it will pass without errors:

```js
var assert = require('assert');
var bitcore = require('bitcore-lib');

var message = new Buffer('52204d20fd0131ae1afd173fd80a3a746d2dcc0cddced8c9dc3d61cc7ab6e966', 'hex');
var reversed = new Buffer('66e9b67acc613ddcc9d8cedd0ccc2d6d743a0ad83f17fd1aae3101fd204d2052', 'hex');                                                                                                                                           

var keyBuffer = new Buffer('16f243e962c59e71e54189e67e66cf2440a1334514c09c00ddcc21632bac9808', 'hex');
var key = bitcore.PrivateKey.fromBuffer(keyBuffer);

var signature1 = bitcore.crypto.ECDSA.sign(message, key).toBuffer().toString('hex');
var signature2 = bitcore.crypto.ECDSA.sign(message, key, 'little').toBuffer().toString('hex');

var secp256k1 = require('secp256k1');

var signature3 = secp256k1.sign(message, keyBuffer);
var signature4 = secp256k1.sign(reversed, keyBuffer);

assert.equal(secp256k1.signatureExport(signature3.signature).toString('hex'), signature1);
assert.equal(secp256k1.signatureExport(signature4.signature).toString('hex'), signature2);
```

File: lib/crypto/ecdsa.js [hunk 1]
```diff
@@ -88,9 +88,10 @@ ECDSA.prototype.deterministicK = function(badrs) {
   var x = this.privkey.bn.toBuffer({
     size: 32
   });
-  k = Hash.sha256hmac(Buffer.concat([v, new Buffer([0x00]), x, this.hashbuf]), k);
+  var hashbuf = this.endian === 'little' ? [].reverse.call(new Buffer(this.hashbuf)) : this.hashbuf
```
Comment (braydonf) [inline lib/crypto/ecdsa.js]: We may want to use `BufferUtil.reverse()` here, since we have benchmarks around it, and use it elsewhere.

**Comment by braydonf** [comment]
LGTM, except nit from above. I think this is also backwards compatibly and should be able to include in `0.13.14`.

**Comment by matiu** [comment]
LGTM. Agree with @braydonf 's suggestion.

Great Job @fanatid and @braydonf checking it.

@crwatkins  this will make Copay's signatures to match other libraries (as mentioned in https://github.com/bitcoin-dot-org/bitcoin.org/pull/888).

**Comment by fanatid** [comment]
Updated.

**Comment by braydonf** [comment]
Thanks! Landed in `0.13.14`

**Pull request merged**